### PR TITLE
ENH: Accept hierarchical fields

### LINKED ISSUE #1468: Wrong field names when dot in names
I would like to retrieve all elements of a PDF form and display for each element it's content. 
When I read fields having a "." in its name, the "key" returned is only the last part of the key, and the value is the value of the last element having this (wrong) key.

## Environment
Windows 11 (Windows-10-10.0.22623-SP0)
Python 3.10
pypdf 3.2.0 / PyPDF2-2.11.2

## Code + PDF
I have a PDF file with the following fields: 
- customer.name
- customer.lastname
- other_field
- company.name

pdf

[fields_with_dots.pdf](https://github.com/py-pdf/PyPDF2/files/10142389/fields_with_dots.pdf)

I want to get all fields and the corresponding values. 
```python
from PyPDF2 import PdfFileReader

if __name__ == "__main__":
    pdf_file_name = "fields_with_dots.pdf"
    with open(pdf_file_name, "rb") as pdfobject:
        pdf = PdfFileReader(pdfobject)
        fields = pdf.getFormTextFields()
        for k, v in fields.items():
            print(f"{k:32} : {v}")
```

### What I expected
```
customer.name               : My Name
customer.lastname         : My Lastname
other_field                      : Hello world!
company.name               : My company
```
### What the result is
```
name                             : My company
lastname                         : My Lastname
other_field                      : Hello world!
```
"customer.name" is identified as "name". "company.name" is also identified as "name". 
The value is the latest found in document : "My Company".
--> value for "customer.name" is lost.

### PR DESCRIPTION
**Pull request by pubpub-zz**
fixes #1468
Add an optional parameter to `get_form_text_fields`

TBC : behavior with multiple fields with same name create sequence _2, _3, ... does not seem to be applied

Pending : Test to be added

### CODE REVIEW

File: pypdf/_reader.py [hunk 1]
```diff
@@ -640,22 +663,26 @@ def _write_field(self, fileobj: Any, field: Any, field_attributes: Any) -> None:
                 # Field attribute is N/A or unknown, so don't write anything
                 pass
 
-    def get_form_text_fields(self) -> Dict[str, Any]:
+    def get_form_text_fields(self, full_qualified_name: bool = False) -> Dict[str, Any]:
```
Comment (MartinThoma) [inline pypdf/_reader.py:666]: Good work in making it backwards compatible :+1: :pray:

File: pypdf/_reader.py [hunk 2]
```diff
@@ -640,25 +663,54 @@ def _write_field(self, fileobj: Any, field: Any, field_attributes: Any) -> None:
...
+            ff = {
+                (field if full_qualified_name else formfields[field]["/T"]): formfields[
+                    field
+                ].get("/V")
+                for field in formfields
+                if formfields[field].get("/FT") == "/Tx"
+            }
+        else:
+            ff = {}
+            for (k, v) in formfields.items():
```
Comment (MartinThoma) [inline pypdf/_reader.py:703]:
```suggestion
            for v in formfields.values():
```

**Comment by pubpub-zz** [comment]
@MartinThoma 
is it sufficient for you ?

File: pypdf/_reader.py [hunk 3]
```diff
@@ -651,25 +674,56 @@ def _write_field(self, fileobj: Any, field: Any, field_attributes: Any) -> None:
...
+
         # Retrieve document form fields
         formfields = self.get_fields()
         if formfields is None:
             return {}
-        return {
-            formfields[field]["/T"]: formfields[field].get("/V")
+        if full_qualified_name:
+            ff = {
+                (field if full_qualified_name else formfields[field]["/T"]): formfields[
```
Comment (MartinThoma) [inline pypdf/_reader.py:706]: That section is weird. At this point `full_qualified_name`should always evaulate to `True`

Comment (pubpub-zz) [reply pypdf/_reader.py:706]: oups...

File: pypdf/_reader.py [hunk 4]
```diff
@@ -651,25 +674,56 @@ def _write_field(self, fileobj: Any, field: Any, field_attributes: Any) -> None:
...
+                for field in formfields
+                if formfields[field].get("/FT") == "/Tx"
+            }
+        else:
+            ff = {}
+            for v in formfields.values():
+                if v.get("/FT") == "/Tx":
+                    ff[indexed_key(cast(str, v["/T"]), ff)] = (
+                        cast(str, v["/V"]) if "/V" in v else None
+                    )
```
Comment (MartinThoma) [inline pypdf/_reader.py:718]: I think that part could be simplified to:

```suggestion
        ff = {}
        for field, value in formfields.items():
            if value.get("/FT") == "/Tx":
                if full_qualified_name:
                    ff[field] = value.get("/V")
                else:
                    ff[indexed_key(cast(str, value["/T"]), ff)] = value.get("/V")
```

**Comment by MartinThoma** [comment]
Overall, the PR looks good to me :-) I will have another look tomorrow, but if it's only those minor stylistic things, I can also fix them after merging :-) 

Good work @pubpub-zz :+1: :pray:

**Pull request merged**